src/powershell/Public/Remove-FinOpsCostExport.ps1

```diff
@@ -0,0 +1,169 @@
+    }
+
+    if ($PSCmdlet.ShouldProcess($Name, 'DeleteCostExport'))
+    {
+
+      # Using the REST API to delete the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "DELETE" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+        # Export deleted successfully
+      }
+      elseif ($httpResponse.StatusCode -eq 404)
+      {
+        # Not found - nothing to delete
+      }
+      else
+      {
+        # Error response describing why the operation failed.
+        throw "Delete Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+      }
+    }
+
+    # Delete associated ingestion data from storage account
+    if ($RemoveData)
+    {
+      # Using the REST API to get the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "GET" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+
+        # Export details retreived
+        $exportDetails = ConvertFrom-Json -InputObject $httpResponse.Content
+        $storageAccountID = $exportDetails.properties.deliveryInfo.destination.resourceId
+
+        # Get-AzStorageAccount -resourcegroupname alz-finopstk -name finopshub255k7ov6asado | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path "/" -Recurse -FetchProperty | Where-Object IsDirectory -ne false | Select-Object Path
+        $resourceGroupName = $storageAccountID.Split('/')[4]
+        $storageAccountName = $storageAccountID.Split('/')[8]
+
+        #Hold on to your hats, deleting all the files in the ingestion scope. Using Az PS module as this should be current.
+        if ($PSCmdlet.ShouldProcess($scope, 'DeleteCostReports'))
+        {
+          Write-Verbose "Resourace Group: $resourceGroupName"
+          Write-Verbose "Storage Account: $storageAccountName"
+          Write-Verbose "Scope: $scope"
+
+          $getFiles = Get-AzStorageAccount -ResourceGroupName $resourceGroupName -Name $storageAccountName | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path $scope -Recurse -FetchProperty
+          if ($getFiles.Count -gt 0)
+          {
+            $getFiles | Remove-AzDataLakeGen2Item -Force
+          }
+        }
+      }
+      elseif ($httpResponse.StatusCode -eq 404)
+      {
+        # Not found - Nothing to delete
+      }
+      else
+      {
+        # Error response describing why the operation failed.
+        throw "Delete ingestion data operation failed with message: '$($httpResponse.Content)'"
+      }
+    }
+
+  }
+  catch
+  {
+    throw $_.Exception.Message
```

Comment:
Unless we are re-wrapping the exception, I would throw then entire exception because it gives us all the info from originating call